src/python_testing/TestConformanceSupport.py

```diff
@@ -0,0 +1,564 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import xml.etree.ElementTree as ElementTree
+
+from conformance_support import ConformanceDecision, ConformanceParseParameters, parse_callable_from_xml
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TestConformanceSupport(MatterBaseTest):
+    @async_test_body
+    async def test_conformance_from_xml(self):
+        # a small feature map
+        feature_names_to_bits = {'AB': 0x01, 'CD': 0x02}
+
+        # none, AB, CD, AB&CD
+        feature_maps = [0x00, 0x01, 0x02, 0x03]
+        has_ab = [False, True, False, True]
+        has_cd = [False, False, True, True]
+
+        attribute_names_to_values = {'attr1': 0x00, 'attr2': 0x01}
+        attribute_lists = [[], [0x00], [0x01], [0x00, 0x01]]
+        has_attr1 = [False, True, False, True]
+        has_attr2 = [False, False, True, True]
+
+        command_names_to_values = {'cmd1': 0x00, 'cmd2': 0x01}
+        cmd_lists = [[], [0x00], [0x01], [0x00, 0x01]]
+        has_cmd1 = [False, True, False, True]
+        has_cmd2 = [False, False, True, True]
+
+        params = ConformanceParseParameters(
+            feature_map=feature_names_to_bits, attribute_map=attribute_names_to_values, command_map=command_names_to_values)
+
+        # Start simple - mandatory, optional, disallowed
+        xml = '<mandatoryConform />'
+        et = ElementTree.fromstring(xml)
+        callable = parse_callable_from_xml(et, params)
```

Comment:
nit: consider `xml_callable` or something else because `callable` is a python built-in:

```
> ptpython
>>> callable
<built-in function callable>
```